src/spec.c

```diff
@@ -795,25 +795,23 @@ RedisModuleString *IndexSpec_GetFormattedKey(IndexSpec *sp, const FieldSpec *fs,
     RedisSearchCtx sctx = {.redisCtx = RSDummyContext, .spec = sp};
     switch (forType) {
       case INDEXFLD_T_NUMERIC:
+      case INDEXFLD_T_GEO: // TODO?? change the name
         ret = fmtRedisNumericIndexKey(&sctx, fs->name);
         break;
       case INDEXFLD_T_TAG:
         ret = TagIndex_FormatName(&sctx, fs->name);
         break;
-      case INDEXFLD_T_GEO:
-        ret = RedisModule_CreateStringPrintf(RSDummyContext, GEOINDEX_KEY_FMT, sp->name, fs->name);
-        break;
       case INDEXFLD_T_FULLTEXT:  // Text fields don't get a per-field index
       default:
         ret = NULL;
         abort();
         break;
     }
+    if (!ret) {
```

Comment:
Can this happened? If yes, do we handle it on all usages? If not lets assert.

Reply:
Not really. put logassert
Should we remove all these functions and use `fs->name` with dict instead?

Reply:
Let's keep it but assert here if it can  not happened